src/picongpu/include/traits/UsesRNG.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2016 Marco Garten, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/mpl/bool.hpp>
+
+namespace picongpu
+{
+namespace traits
+{
+
+/** Checks if an object requires the RNG
+ *
+ * @tparam T_Object any object (class or typename)
+ *
+ * This struct must define
+ * ::type (boost::mpl::bool_<>)
+ */
+template<typename T_Object>
+struct UsesRNG
```

Comment:
Note that boost `integral_constant`s **are** in fact nullary metafunctions. See their definition:

    [...]struct integral_constant : public mpl::integral_c<T, val>[...]

This is why the above use of `boost::is_unsigned` works. This is defined [here](http://www.boost.org/doc/libs/1_53_0/libs/type_traits/doc/html/boost_typetraits/reference/integral_constant.html) and also in [C++11](http://www.cplusplus.com/reference/type_traits/true_type/)

So as a matter of fact you can (and should) inherit from `boost::true_type`/`boost::false_type` which was my original statement :)